core/iwasm/libraries/lib-wasi-threads/lib_wasi_threads_wrapper.c

```diff
@@ -114,21 +164,28 @@ thread_spawn_wrapper(wasm_exec_env_t exec_env, uint32 start_arg)
     }
 
     thread_start_arg->thread_id = thread_id = allocate_thread_id();
+    if (thread_id < 0) {
+        LOG_ERROR("Failed to get thread identifier");
+        goto thread_spawn_fail;
```

Comment:
the label is a bit confusing; we didn't get to spawning thread yet, so I'd expect we should use a `tid_allocation_fail` label instead (and probably change the implementation of that below).